### PR TITLE
Import GleanTest.kt from glean-ac

### PR DESCRIPTION
**Pull request by Dexterp37**
Import the tests over from [this version GleanTest.kt](https://raw.githubusercontent.com/mozilla-mobile/android-components/8ef0d4b3247255ca6c456e9b5ab61338bc109209/components/service/glean/src/test/java/mozilla/components/service/glean/GleanTest.kt) from glean-ac.

**Note**: the first commit simply overwrites the current test file we own.

### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - `make test` runs without emitting any warnings
  - `make lint` runs without emitting any errors
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry or an explanation of why it does not need one
  - Any breaking changes to language binding APIs are noted explicitly

### CODE REVIEW

File: glean-core/android/src/test/java/mozilla/telemetry/glean/GleanTest.kt [hunk 1]
```diff
@@ -53,4 +103,361 @@ class GleanTest {
...
+            val baselineTimespanMetrics = baselineMetricsObject.getJSONObject("timespan")!!
+            assertEquals(1, baselineTimespanMetrics.length())
+            assertNotNull(baselineTimespanMetrics.get("glean.baseline.duration"))
+        } finally {
+            server.shutdown()
+            lifecycleRegistry.removeObserver(gleanLifecycleObserver)
+        }
+    }
+
+    @Ignore("This is currently failing")
```
Comment (badboy) [inline glean-core/android/src/test/java/mozilla/telemetry/glean/GleanTest.kt]: Do we have a bug for that?

File: glean-core/android/src/test/java/mozilla/telemetry/glean/GleanTest.kt [hunk 2]
```diff
@@ -53,4 +103,361 @@ class GleanTest {
...
+
+    // glean-ac test removed.
+    // `client_id and first_run_date metrics should be copied from the old location` was here.
+    // 1539480 BACKWARD COMPATIBILITY HACK that is not needed anymore.
+
+    // glean-ac test removed.
+    // `client_id and first_run_date metrics should not override new location` was here.
+    // 1539480 BACKWARD COMPATIBILITY HACK that is not needed anymore.
+
+    @Ignore("This should probably be implemented in Rust")
```
Comment (badboy) [inline glean-core/android/src/test/java/mozilla/telemetry/glean/GleanTest.kt]: Uh, this is important. Do we have a bug already?

File: glean-core/android/src/test/java/mozilla/telemetry/glean/GleanTest.kt (hunk 1 above)
Comment (Dexterp37) [reply glean-core/android/src/test/java/mozilla/telemetry/glean/GleanTest.kt]: I filed [bug 1554081](https://bugzilla.mozilla.org/show_bug.cgi?id=1554081) for it.

File: glean-core/android/src/test/java/mozilla/telemetry/glean/GleanTest.kt (hunk 2 above)
Comment (Dexterp37) [reply glean-core/android/src/test/java/mozilla/telemetry/glean/GleanTest.kt]: Right, filed [bug 1554082](https://bugzilla.mozilla.org/show_bug.cgi?id=1554082) for it.

**Pull request merged**